python/dgl/transform.py

```diff
@@ -1367,4 +1367,125 @@ def as_immutable_graph(hg):
     g.edata.update(hg.edata)
     return g
 
+def compose(graphs):
+    """Return a composition of the given list of graphs.
+
+    Given the list of graphs with the same set of nodes, ``compose`` will return a new
+    graph with the same set of nodes with all the edges from the list of graphs.
+
+    All edges from all the graphs will be added.
+
+    If the graphs are heterogeneous, the graphs must have the same metagraph.  All edges from
+    all the graphs will be added for each edge type.
+
+    The edges in the first graph will be added first, followed by the edges from the second graph,
+    etc.
+
+    Notes
+    -----
+    All graphs must have the same set of nodes and metagraphs.  An error will be raised otherwise.
+
+    Parameters
+    ----------
+    graphs : list[DGLHeteroGraph]
+        The list of graphs
+
+    Returns
+    -------
+    DGLHeteroGraph
+        The composed graph.
+
+    Examples
+    --------
+    >>> g1 = dgl.graph(([0, 1, 2], [1, 0, 2]), num_nodes=5)
+    >>> g2 = dgl.graph(([1, 3, 3], [0, 2, 3]), num_nodes=5)
+    >>> g = dgl.compose([g1, g2])
+    >>> g.all_edges(order='eid')
+    (tensor([0, 1, 2, 1, 3, 3]), tensor([1, 0, 2, 0, 2, 3]))
+    """
+    # Exposes the UnionCOO/UnionCSR functions to users here
+    pass
+
+def add_reverse(g):
+    """Return a new graph with reverse edges added to the given homogeneous graph.
+
+    For each edge with ID ``e`` connecting from node ``u`` to node ``v``, a new
+    edge with ID ``e + |E|`` connecting from node ``v`` to node ``u`` is added,
+    where ``|E|`` is the number of edges in the original graph.
+
+    Notes
+    -----
+    The graph must be homogeneous (i.e. with one node type and one edge type).
+
+    Parameters
+    ----------
+    g : DGLHeteroGraph
```

Comment:
I remember Minjie said that they will all be called DGLGraph in the future?

Reply:
Hmm because people will actually use nightly build I guess I'll keep this name for now.  When we truly merge DGLGraph and DGLHeteroGraph I guess we can simply replace all occurrences of DGLHeteroGraph to DGLGraph with `sed`.